Core/Core/Data/Model/Data_PrimaryEnrollment.swift

```diff
@@ -0,0 +1,269 @@
+        public let mode: String?
+        public let isActive: Bool?
+        public let course: DashboardCourse?
+        public let certificate: DataLayer.Certificate?
+        public let courseModes: [CourseMode]?
+        public let courseStatus: CourseStatus?
+        public let progress: Progress?
+        public let courseAssignments: CourseAssignments?
+        
+        enum CodingKeys: String, CodingKey {
+            case auditAccessExpires = "audit_access_expires"
+            case created
+            case mode
+            case isActive = "is_active"
+            case course
+            case certificate
+            case courseModes = "course_modes"
+            case courseStatus = "course_status"
+            case progress = "course_progress"
+            case courseAssignments = "course_assignments"
+        }
+        
+        public init(
+            auditAccessExpires: Date?,
+            created: String?,
+            mode: String?,
+            isActive: Bool?,
+            course: DashboardCourse?,
+            certificate: DataLayer.Certificate?,
+            courseModes: [CourseMode]?,
+            courseStatus: CourseStatus?,
+            progress: Progress?,
+            courseAssignments: CourseAssignments?
+        ) {
+            self.auditAccessExpires = auditAccessExpires
+            self.created = created
+            self.mode = mode
+            self.isActive = isActive
+            self.course = course
+            self.certificate = certificate
+            self.courseModes = courseModes
+            self.courseStatus = courseStatus
+            self.progress = progress
+            self.courseAssignments = courseAssignments
+        }
+    }
+    
+    // MARK: - CourseStatus
+    struct CourseStatus: Codable {
+        public let lastVisitedModuleID: String?
+        public let lastVisitedModulePath: [String]?
+        public let lastVisitedBlockID: String?
+        public let lastVisitedUnitDisplayName: String?
+
+        enum CodingKeys: String, CodingKey {
+            case lastVisitedModuleID = "last_visited_module_id"
+            case lastVisitedModulePath = "last_visited_module_path"
+            case lastVisitedBlockID = "last_visited_block_id"
+            case lastVisitedUnitDisplayName = "last_visited_unit_display_name"
+        }
+    }
+    
+    // MARK: - CourseAssignments
+    struct CourseAssignments: Codable {
+        public let futureAssignments: [Assignment]?
+        public let pastAssignments: [Assignment]?
+        
+        enum CodingKeys: String, CodingKey {
+            case futureAssignments = "future_assignments"
+            case pastAssignments = "past_assignments"
+        }
+        
+        public init(futureAssignments: [Assignment]?, pastAssignments: [Assignment]?) {
+            self.futureAssignments = futureAssignments
+            self.pastAssignments = pastAssignments
+        }
+    }
+    
+    // MARK: - Assignment
+    struct Assignment: Codable {
```

Comment:
Saeed, since there are no other Assignment structs in the codebase, there isn't a need to rename it to CourseAssignment. Keeping it as Assignment keeps the code simpler and avoids unnecessary verbosity. If we encounter a need for another Assignment struct in the future, we can revisit this decision and consider renaming at that time.

Reply:
For the time being we can go with this approach. The naming seems a bit odd from the other supporting structs names.